lib/ansible/cli/__init__.py

```diff
@@ -605,8 +604,8 @@ def parse(self):
                     skip_tags.add(tag.strip())
             self.options.skip_tags = list(skip_tags)
 
-        # process inventory options
-        if hasattr(self.options, 'inventory'):
+        # process inventory options except for CLIs that require their own processing
+        if hasattr(self.options, 'inventory') and self.__class__.__name__ not in ('PullCLI'):
```

Comment:
Should this be a flag set within the class rather than a check for a hard-coded class name?

Reply:
good point